### PR TITLE
Fix dead link in rdm-spreadsheets

### LINKED ISSUE #1334: [BUG] dead link on the rdm-spreadsheets page
### Bug description

- dead link in [Using spreadsheets for research data](https://the-turing-way.netlify.app/reproducible-research/rdm/rdm-spreadsheets.html)
- The following is not a valid DOI.

```
Some of these mishaps are not unique to spreadsheets, but [many are](https://doi.org/10.1186/s13059-016-104).
```

I can happily fix that but I am not sure what the target is supposed to be. :-)

On a related note: maybe I missed it but there seems to be no automatic checks for dead links.

In case you are looking for an easy way to set one up I have started using this one and can give a hand setting it up.

https://github.com/marketplace/actions/markdown-link-check

### PR DESCRIPTION
**Pull request by JimMadge**
### Summary

The 'Using spreadsheets for research data' page contained a dead link. It appears that the link became truncated in a previous commit. This PR restores that original link and adds a second similar and relevant example.

Fixes #1334 

### List of changes proposed in this PR (pull-request)

- Restore url from `https://doi.org/10.1186/s13059-016-104` to `https://doi.org/10.1186/s13059-016-1044-7` as it was before commit 27314d4ce5a9bc53327f8e013300df2497d234ff
- Add a second, similar example of spreadsheets introducing unanticipated errors. This example was suggested by @malvikasharan in #1334

### What should a reviewer concentrate their feedback on?

- [x] Everything looks ok?
- [ ] The two hyperlinks to examples are on adjacent words, is this clear and fitting with the style?

### Acknowledging contributors

- [x] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Comment by malvikasharan** [comment]
Thank you @JimMadge for fixing these links. I will ask new contributors to review this PR.

File: book/website/reproducible-research/rdm/rdm-spreadsheets.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Spreadsheets are convenient and easy-to-use tools for organising information int
 However, one should use them with caution, as the use of an inappropriate spreadsheet is a major cause of mistakes in the data analysis workflow.
 There is a collection of [horror-stories](http://www.eusprig.org/horror-stories.htm) that tells how the use of spreadsheets can ruin analysis-based studies due to unexpected behaviour of the spreadsheet or error-prone
 editing processes.
-Some of these mishaps are not unique to spreadsheets, but [many are](https://doi.org/10.1186/s13059-016-104).
+Some of these mishaps are not unique to spreadsheets, but [many](https://doi.org/10.1186/s13059-016-1044-7) [are](https://doi.org/10.1186/1471-2105-5-80).
```
Comment (paulowoicho) [inline book/website/reproducible-research/rdm/rdm-spreadsheets.md:9]:
```suggestion
Some of these mishaps are not unique to spreadsheets, but many, such as [this](https://doi.org/10.1186/s13059-016-1044-7) and [this](https://doi.org/10.1186/1471-2105-5-80), are.
```

**Review by paulowoicho — COMMENTED** [review]
Hi @JimMadge! Thank you for fixing these! I didn't think it was clear that there were two separate links because the words were adjacent, so I suggested to re-word the sentence a little to make this more apparent.

**Review by paulowoicho — APPROVED** [review]

**Comment by malvikasharan** [comment]
Great! please go ahead and merge it if everything's ok @paulowoicho. I have already added Jim to the contributor's table.

**Pull request merged**